lib/einhorn/client.rb

```diff
@@ -26,7 +26,14 @@ def self.serialize_message(message)
 
       def self.deserialize_message(line)
         serialized = line.gsub(/%(25|0A)/, "%25" => "%", "%0A" => "\n")
-        YAML.load(serialized)
+        # YAML.load(serialized)
+        if RUBY_VERSION >= "2.6.0"
```

Comment:
Haven't been able to run this because I'm failing to get ruby 2.5 reinstalled but checking the method parameters of `safe_load` should show if it uses `permitted_parameters`

```suggestion
        yaml_permitted_classes = if YAML.respond_to?(:safe_load) && YAML.method(:safe_load).parameters.any?{ |pair| pair[1] == :permitted_classes }
        if yaml_permitted_classes
```